src/strategies/gadget.rs

```diff
@@ -365,44 +375,45 @@ mod tests {
             });
 
             composer.add_dummy_constraints();
-            //let pub_inp = composer.public_inputs().clone();
-            (composer, vec![BlsScalar::zero()])
+            vec![BlsScalar::zero()]
         }
 
         // Setup OG params.
         let public_parameters = PublicParameters::setup(CAPACITY, &mut rand::thread_rng()).unwrap();
         let (ck, vk) = public_parameters.trim(CAPACITY).unwrap();
-        let domain = EvaluationDomain::new(CAPACITY).unwrap();
 
         let (i, o) = hades();
-        let (mut composer, pi) = new_composer(i, o);
-        let mut transcript = gen_transcript();
-        // Preprocess circuit
-        let circuit = composer.preprocess(&ck, &mut transcript, &domain);
-
-        // Prove
-        let proof = composer.prove(&ck, &circuit, &mut transcript.clone());
-
-        // Verify
-        assert!(proof.verify(&circuit, &mut transcript.clone(), &vk, pi.as_slice()));
-
+        // Proving
+        let mut prover = Prover::new(b"hades_gadget_tester");
+        let pi = hades_gadget_tester(i, o, prover.mut_cs());
+        prover.preprocess(&ck).unwrap();
+        let proof = prover.prove(&ck).unwrap();
+
+        // Verifying
+        let mut verifier = Verifier::new(b"hades_gadget_tester");
+        let _ = hades_gadget_tester(i, o, verifier.mut_cs());
+        verifier.preprocess(&ck).unwrap();
```

Comment:
Ditto.

Reply:
WDYM?

Reply:
**Ditto** – _the same thing again (used in lists and accounts and often indicated by a ditto mark under the word or figure to be repeated)._